File: 1047_remove_all_adjacent_duplicates_in_string/remove_all_adjacent_duplicates_in_string.c
char * removeDuplicates(char * S){
    int size = strlen(S);
    char *ans = malloc(sizeof(char) * (size + 1));
    int position = 0;
    for (int i = 0; i < size; i++) {
        if (position > 0 && ans[position-1] == S[i])
            position--;
        else
            ans[position++] = S[i];
    }
    ans[position] = '\0';
    return ans;
}

File: 1047_remove_all_adjacent_duplicates_in_string/remove_all_adjacent_duplicates_in_string.java
class Solution {
    public String removeDuplicates(String S) {
        StringBuffer stack = new StringBuffer();
        int top = -1;
        for (int i = 0; i < S.length(); ++i) {
            char ch = S.charAt(i);
            if (top >= 0 && stack.charAt(top) == ch) {
                stack.deleteCharAt(top);
                --top;
            } else {
                stack.append(ch);
                ++top;
            }
        }
        return stack.toString();
    }
}

char* removeDuplicates(char* S) {
    int n = strlen(S);
    char* stk = malloc(sizeof(char) * (n + 1));
    int retSize = 0;
    for (int i = 0; i < n; i++) {
        if (retSize > 0 && stk[retSize - 1] == S[i]) {
            retSize--;
        } else {
            stk[retSize++] = S[i];
        }
    }
    stk[retSize] = '\0';
    return stk;
}

